File: app/src/main/java/fr/nextgear/mesentretiensmoto/core/database/TableContracts.java
package fr.nextgear.mesentretiensmoto.core.database;

/**
 * Created by adrien on 14/06/2017.
 */

public interface TableContracts {

    interface Bike{
        String TABLE_NAME = "bike_table";
        String ID = "id_bike";
        String NAME = "name_bike";
        String MAINTENANCES = "maintenances_bike";

    }

    interface Maintenance{
        String TABLE_NAME = "maintenance_table";
        String ID = "id_maintenance";
        String NAME = "name_maintenance";
        String NB_HOURS = "nb_hours_maintenance";
        String DATE = "date_maintenance";
        String IS_DONE = "maintenance_is_done";
        String BIKE_ID = "maintenance_bike_id";
    }

    interface MaintenancePrevision{
        String TABLE_NAME = "maintenance_prevision_table";
        String ID = "id_maintenance_prevision";
        String MAINTENANCE = "maintenance_maintenance_prevision";
    }
}

File: app/src/main/java/fr/nextgear/mesentretiensmoto/core/database/TableContracts.kt
package fr.nextgear.mesentretiensmoto.core.database

import android.provider.BaseColumns

/**
 * Created by adrien on 14/06/2017.
 */

object TableContracts {

    const val DB_NAME = "maintenances_bike.sqlite"
    const val DB_VERSION = 12

    object Bike : BaseColumns {
        const val TABLE_NAME = "bike"
        const val ID = "id_bike"
        const val NAME = "name_bike"
        const val MAINTENANCES = "maintenances_bike"
        const val REF_STR = "reference_bike"
    }

    object Maintenance : BaseColumns {
        const val TABLE_NAME = "maintenance"
        const val ID = "id_maintenance"
        const val NAME = "name_maintenance"
        const val NB_HOURS = "nb_hours_maintenance"
        const val DATE = "date_maintenance"
        const val IS_DONE = "maintenance_is_done"
        const val BIKE_ID = "maintenance_bike_id"
        const val REF_STR = "reference_maintenance"
    }

}
